### DISCUSSION TITLE
深度终端：【建议】1. 增加 --new-tab类似的参数，用于打开新标签页。

### DISCUSSION THREAD

**Discussion by MicroOps-cn** [linuxdeepin/developer-center#2861] [Features Request & Ideas | 特性请求 & 头脑风暴]
【建议】1. 增加 --new-tab类似的参数，用于打开新标签页。

**Comment by ch593030323** [comment]
感谢您的建议，我们会合理考虑此功能

**Comment by meiyixiang** [comment]
你好可以说的具体点吗？我不能够理解你的意思。

**Reply by BLumia** [reply]
感觉意思应该是希望 `deepin-terminal --new-tab` 的方式来创建一个终端新标签页

**Reply by MicroOps-cn** [reply]
> 你好可以说的具体点吗？我不能够理解你的意思。
我现在需要通过一个第三方工具管理密码，配合deepin-terminal来实现一键登录，打开的时候每次都是打开一个新窗口，很不方便，我是希望打开的时候可以放入同一窗口内
参考: #2813

**Reply by MicroOps-cn** [reply]
我自己现在是在我fork的代码内实现了这个功能
https://github.com/MicroOps-cn/deepin-terminal/commit/b78bad3116838bb648ff75fb6c6418b962b49d37

**Comment by meiyixiang** [comment]
我理解为我们在终端已经提供了新建标签的快捷按钮，这个功能没有什么意义

**Reply by MicroOps-cn** [reply]
> 我理解为我们在终端已经提供了新建标签的快捷按钮，这个功能没有什么意义

那比如现在的-e (--execute)参数，的意义是什么呢？既然我都打开了终端，我直接输入命令不一样吗？
既然提供了这个功能，应该是有人有这种特殊需求吧。比如通过其他应用来调用这个终端运行一个命令。

**Comment by meiyixiang** [comment]
明白你的意思，你的建议我已经反馈给相关的产品经理。

**Comment by NOTOKComputers** [comment]
对我来说，我希望我的linux桌面里能集成快捷键，一键开启深度终端里的一个new-tab, 而不是一个window。 

类似于win+enter可以触发

`deepin-terminal --new-tab` 命令的这种形式。

这样每次命令都不会触发新的deepin-terminal窗口，而只是既有窗口的一个新tab。这样方便管理alt-tab里显示的窗口数量。

---
目前我使用快捷键触发类似 `deepin-terminal -e "/home/user/<你的用户名>/bin/tmux-login-session"` 命令的方式缓解，哪怕多弹出了一些终端，由于都套了相同的tmux-session，退出的时候其他终端也会退出，管理窗口稍微方便了一点。

`~/bin/tmux-login-session`脚本：

```bash
#!/bin/sh
if tmux has-session -t "common"; then
	exec tmux a -t "common"
else
	. ~/.profile
	exec sh -c "tmux new-s -d -s common && tmux new-s -s new"
fi
```
